src/cpp/src/EXAMPLE/dumpmctree-dot.cc

```diff
@@ -0,0 +1,809 @@
+        double pz = mc->getMomentum()[2];
+        labels<<p2idx[mc]<<"[label=<"<<label<<"<BR/>"<<std::fixed<<std::setprecision(2)<<dToIP<<" mm<BR/>"<<pt<<" | "<<pz<<" GeV"<<">";
+        if ( isBeforeHadronisation(mc) ) labels<<" style=\"filled\" fillcolor=\"burlywood4\"";
+        else if ( mc->isOverlay() && mc->isCreatedInSimulation() ) labels<<" style=\"filled\" fillcolor=\"darkorange4\"";
+        else if ( mc->isOverlay() ) labels<<" style=\"filled\" fillcolor=\"dimgray\"";
+        else if ( mc->isCreatedInSimulation() ) labels<<" style=\"filled\" fillcolor=\"darkorange\"";
+        labels<<"];"<<std::endl;
+    }
+
+    //create dot file with a header
+    std::ofstream outfile;
+    outfile.open("dumpmctree_tmp.dot");
+    outfile<<"digraph {"<<std::endl;
+    outfile<<"    rankdir=TB;"<<std::endl;
+    outfile<<nodes.str()<<std::endl;
+    outfile<<labels.str()<<std::endl;
+    outfile<<"}"<<std::endl;
+}
+
+
+constexpr static auto usage = R"USAGE(usage: dumpmctree [-s] [-p] [-o] filename eventNumber)USAGE";
+constexpr static auto help = R"HELP(
+OPTIONS:
+    -s:             Draw particles created in simulation highlighted in orange
+    -p:             Draw parton shower particles above pythia hadronisation string/cluster (pdg = 92/91).
+                    E.g. (Z/W/H/quarks/gluons/etc). Highlighted in burlywood
+    -o:             Draw overlay particles. Highlighted in gray
+INPUT:
+    filename:       input slcio file with an event to draw
+    eventNumber:    event number to draw
+)HELP";
+
+
+/** dump MC table of the event from the slcio file in the neat graphviz tree representation.
+*/
+int main(int argc, char** argv ){
+    //options (optional)
+    bool drawSimulated = false;
+    bool drawParton = false;
+    bool drawOverlay = false;
+
+    //inputs (required)
+    char* filename ;
+    int eventNumber{};
+
+    if ( (argc < 3) ||
+         (argv[argc-1] == NULL) || (argv[argc-2] == NULL) ||
+         (argv[argc-1][0] == '-') || (argv[argc-2][0] == '-') ) {
+        // there is NO input
+        std::cerr << usage << std::endl;
+        std::cerr << help << std::endl;
+        return 1;
+    }
+    else {
+        // there is an input
+        filename = argv[argc-2];
+        eventNumber = atoi( argv[argc-1] );
+    }
+
+    // Retrieve the options:
+    while ( true ){
+        switch ( getopt(argc, argv, "spoh") ) {
+            case 's':
+                drawSimulated = true;
+                continue;
+            case 'p':
+                drawParton = true;
+                continue;
+            case 'o':
+                drawOverlay = true;
+                continue;
+            case '?':
+                std::cerr << usage << std::endl;
+                std::cerr << help << std::endl;
+                return 1;
+            case 'h':
+                std::cerr << usage << std::endl;
+                std::cerr << help << std::endl;
+                return 1;
```

Comment:
```suggestion
                return 0;
```
Simply because the user wanted the help, so this can be considered as success at this point.